x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -436,17 +492,20 @@ public static RealmRef randomRealmRef(boolean underDomain, boolean includeSingle
             KerberosRealmSettings.TYPE,
             randomAlphaOfLengthBetween(3, 8)
         );
-        final Supplier<String> randomRealmTypeSupplier;
-        if (includeSingletons) {
-            randomRealmTypeSupplier = randomAllRealmTypeSupplier;
+        if (includeInternal) {
+            return randomAllRealmTypeSupplier;
         } else {
-            randomRealmTypeSupplier = () -> randomValueOtherThanMany(
-                value -> value.equals(FileRealmSettings.TYPE) || value.equals(NativeRealmSettings.TYPE),
+            return () -> randomValueOtherThanMany(
+                value -> value.equals(FileRealmSettings.TYPE) || value.equals(NativeRealmSettings.TYPE) || value.equals("reserved"),
                 randomAllRealmTypeSupplier
             );
```

Comment:
The way how `reserved` works in these test methods implies that it is _not_ the true `reserved` realm. For an example, it can have a random name. Or it can be part of a domain. Therefore, I don't think we should exclude it here for internal realms.